### PR TITLE
Remove `child-process-promise` dependency

### PR DESCRIPTION
**Pull request by mrm007**
And replace it with `util.promisify`ed APIs.

It's only used in tests so no impact to consumers.

### CODE REVIEW

File: test-utils/runSkuScriptInDir.js [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 const runSkuScriptInDir = async (script, cwd, args = []) => {
   // When starting a dev server, return a hook to the running process
   if (/^(start|storybook|serve)/.test(script)) {
-    return gracefulSpawn(skuBin, [script, ...args], { stdio: 'inherit', cwd });
+    return gracefulSpawn(skuBin, [script, ...args], {
+      stdio: 'inherit',
+      cwd,
+      env: {
+        ...process.env,
+        STORYBOOK_DISABLE_TELEMETRY: 1,
```
Comment (mrm007) [inline test-utils/runSkuScriptInDir.js:16]: https://storybook.js.org/docs/react/configure/telemetry

File: test-utils/runSkuScriptInDir.js [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
-      //
-      // See node v10: https://nodejs.org/docs/latest-v10.x/api/child_process.html#child_process_child_process_exec_command_options_callback
-      // See node v12: https://nodejs.org/docs/latest-v12.x/api/child_process.html#child_process_child_process_exec_command_options_callback
-      maxBuffer: 1024 * 1024,
+      // Increased from 1024 * 1024 because Storybook can produce very large outputs.
+      // https://nodejs.org/docs/latest-v18.x/api/child_process.html#child_process_child_process_exec_command_options_callback
+      maxBuffer: 5 * 1024 * 1024,
     });
+    await promise;
+    return { child: promise.child };
```
Comment (mrm007) [inline test-utils/runSkuScriptInDir.js:32]: This one is a little weird. Returning the promise we only get access to `stderr` and `stdout` 🤷 https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/child_process.d.ts#L961-L967

Comment (askoufis) [reply test-utils/runSkuScriptInDir.js:32]: Some tests rely on the exit code 🤔 https://github.com/seek-oss/sku/actions/runs/5386488039/jobs/9776605749#step:13:952

Comment (mrm007) [reply test-utils/runSkuScriptInDir.js:32]: Yep, missed a few

**Review by askoufis — APPROVED** [review]

**Pull request merged**